### PR TITLE
feat(build production): introduce the production build

### LINKED ISSUE #41: Production build
```
ng build --environment=production
```

generate dist with dev versions of angular and systemjs, and my app files with .ts and source maps.

### PR DESCRIPTION
**Pull request by cironunes**
For production build:
- removed live reload references (still not 100%)
- stopped copying tests and .ts files + tsconfig.json

Missing:
- [ ] enableProdMode()
- [ ] add a CNAME file
- [ ] inlining of CSS and HTML resources (#296)
- [x] .js minification

Closes #41

### CODE REVIEW

**Comment by cironunes** [comment]
I'm not sure if this is the best approach. I was trying to create the `build` command and extend the `ember-cli`, but I got stuck at some point and just decided to deliver something so you guys can give feedback.

/cc @angular/angular-cli-team

**Comment by hansl** [comment]
Looks good but Angular2App was changed significantly this weekend

**Comment by cironunes** [comment]
I'll update it accordingly today.

**Comment by hansl** [comment]
Cool. LGTM :)

We'll need to improve a LOT but this is a good first step.

**Comment by hansl** [comment]
You'll have to rebase again. We're getting closer though. I can help with the rest :)

**Comment by cironunes** [comment]
I'll do that soon :)

File: lib/broccoli/angular2-app.js [hunk 1]
```diff
@@ -227,6 +176,166 @@ Angular2App.prototype.index = function () {
...
+
+
+/**
+ Returns the `excludeDotfiles` tree
+
+ @private
+ @method _getDotfilesTree
+ @return {Tree} The dotfiles exclusion tree.
+ */
+Angular2App.prototype._getDotfilesTree = function () {
```
Comment (hansl) [inline lib/broccoli/angular2-app.js]: The purpose of this is more `_getPublicTree`.

File: lib/broccoli/angular2-app.js [hunk 2]
```diff
@@ -227,6 +176,166 @@ Angular2App.prototype.index = function () {
...
   return configReplace(index, {
     files: [htmlName],
-    patterns: this._configReplacePatterns()
+    patterns: this._getReplacePatterns()
+  });
+};
+
+
+/**
+ Returns the source root dir tree
```
Comment (hansl) [inline lib/broccoli/angular2-app.js]: When using the `/** ... */` notation, every line should be prefixed with ` * `.

Like this:

```
/**
 * Returns the source root dir tree.
 * @private
 * @return {Tree} The tree containing all source files.
 */
```

Please do this for all the comments that match this format. Also, end your sentences with a period (`.`).

**Comment by hansl** [comment]
Very good. A few nits. Could you also squash your commits into 1?

I think this is very close to being done.

**Comment by hansl** [comment]
LGTM.

**Pull request merged**